apisix/plugin.lua

```diff
@@ -150,10 +150,13 @@ local function load_plugin(name, plugins_list, plugin_type)
     core.table.insert(plugins_list, plugin)
 
     if plugin.init then
-        plugin.init()
+        if  type(plugin.init) == "function" then
+            plugin.init()
```

Comment:
Why do we need this?

Reply:
check type  of `plugin.init`.   customer plugin may  define it  as other  data  types

Reply:
We can add it in https://github.com/apache/apisix/blob/761c34a292d187fbc49ddb06f1be0d7790df3815/apisix/plugin.lua#L114 and reject invalid plugin